src/App.js

```diff
@@ -1,7 +1,42 @@
 import logo from './logo.svg';
+import { useState } from 'react';
 import './App.css';
+import foodsData from './foods.json';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm';
+import SearchBar from './components/SearchBar';
 
 function App() {
+  const [foods, setAllFoods] = useState(foodsData);
+  const [foodMaster,setFoodMaster] = useState(foodsData)
+
+  function addNewFood(newFood){
+    const updatedFood = [...foods,newFood]
+    setAllFoods(updatedFood)
+    setFoodMaster(updatedFood)
+  }
+
+  const searchFood = (str) => {
+    let searchedFood
+
+    if(str === '' ){
+      searchedFood = foodMaster
+    } else {
+      searchedFood = foodMaster.filter((food) =>{
+        return food.name.toLowerCase() === str.toLowerCase() 
```

Comment:
It would be better to use the include method so actually would start looking for the string and not specting to be exacting the same 

```suggestion
        return food.name.toLowerCase().includes(str.toLowerCase())
```